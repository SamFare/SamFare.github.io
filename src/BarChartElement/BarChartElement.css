.barChartContainer {
    width: 100%; 
    align-content: center;
    margin-top: 10px;
}


.barChart {
    margin: auto;
    width: 75%;
    height: 200px;
}