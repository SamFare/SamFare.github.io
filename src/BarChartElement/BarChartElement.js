import React from "react";
import PropTypes from "prop-types";

import { BarChart, CartesianGrid, XAxis, YAxis,Tooltip, Legend, Bar } from "recharts";

import css from "./BarChartElement.css";



export default class BarChartElement extends React.Component{
  constructor(props) {
    super(props);
    this.data = props.data;
  }

  static get propTypes() { 
    return {
      data: PropTypes.array,
    };
  }

  render(){
    return (  
      <div className={css.barChartContainer}>  
        <BarChart className={css.barChart} width={900} height={400} layout={"vertical"} data={this.data} margin={{top: 5, right: 30, left: 20, bottom: 5}}>
          <CartesianGrid strokeDasharray="3 3" />
          <XAxis dataKey="skillIQ" type="number" />
          <YAxis interval={0} dataKey="skill" type="category" />
          <Tooltip />
          <Legend />
          <Bar dataKey="skillIQ" fill="#8884d8" />
        </BarChart>
      </div>
    );
  }
}