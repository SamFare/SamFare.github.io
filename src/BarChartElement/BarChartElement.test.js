import React from "react";
import BarChartElement from "./BarChartElement.js";
import renderer from "react-test-renderer";

import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

let context = {};
 
const data = [
  {skill: "ES6", skillIQ: 137},
  {skill: "c#", skillIQ: 95},
  {skill: "React", skillIQ: 200},
  {skill: "CSS", skillIQ: 150}
];

describe("a bar chart element", () => {
  beforeEach(() => {
    context.barChartElement = renderer.create(
      <BarChartElement data={data}/>
    );
  });

  it("it has the image that was passed in", () => {
    expect(context.barChartElement).toMatchSnapshot();
  });
});