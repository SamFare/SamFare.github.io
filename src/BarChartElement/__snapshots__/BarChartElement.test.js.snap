// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a bar chart element it has the image that was passed in 1`] = `
<div>
  <div
    className="recharts-wrapper"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "cursor": "default",
        "height": 400,
        "position": "relative",
        "width": 900,
      }
    }
  >
    <svg
      className="recharts-surface"
      height={400}
      version="1.1"
      viewBox="0 0 900 400"
      width={900}
    >
      <clipPath
        id="recharts2-clip"
      >
        <rect
          height={360}
          width={790}
          x={80}
          y={5}
        />
      </clipPath>
      <g
        className="recharts-cartesian-grid"
      >
        <g
          className="recharts-cartesian-grid-horizontal"
        >
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={80}
            x2={870}
            y={5}
            y1={50}
            y2={50}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={80}
            x2={870}
            y={5}
            y1={140}
            y2={140}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={80}
            x2={870}
            y={5}
            y1={230}
            y2={230}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={80}
            x2={870}
            y={5}
            y1={320}
            y2={320}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={80}
            x2={870}
            y={5}
            y1={5}
            y2={5}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={80}
            x2={870}
            y={5}
            y1={365}
            y2={365}
          />
        </g>
        <g
          className="recharts-cartesian-grid-vertical"
        >
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={80}
            x2={80}
            y={5}
            y1={5}
            y2={365}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={277.5}
            x2={277.5}
            y={5}
            y1={5}
            y2={365}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={475}
            x2={475}
            y={5}
            y1={5}
            y2={365}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={672.5}
            x2={672.5}
            y={5}
            y1={5}
            y2={365}
          />
          <line
            fill="none"
            height={360}
            stroke="#ccc"
            strokeDasharray="3 3"
            width={790}
            x={80}
            x1={870}
            x2={870}
            y={5}
            y1={5}
            y2={365}
          />
        </g>
      </g>
      <g
        className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
      >
        <line
          className="recharts-cartesian-axis-line"
          fill="none"
          height={30}
          stroke="#666"
          width={790}
          x={80}
          x1={80}
          x2={870}
          y={365}
          y1={365}
          y2={365}
        />
        <g
          className="recharts-cartesian-axis-ticks"
        >
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={790}
              x={80}
              x1={80}
              x2={80}
              y={365}
              y1={371}
              y2={365}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={790}
              x={80}
              y={373}
            >
              <tspan
                dy="0.71em"
                x={80}
              >
                0
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={790}
              x={80}
              x1={277.5}
              x2={277.5}
              y={365}
              y1={371}
              y2={365}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={790}
              x={277.5}
              y={373}
            >
              <tspan
                dy="0.71em"
                x={277.5}
              >
                50
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={790}
              x={80}
              x1={475}
              x2={475}
              y={365}
              y1={371}
              y2={365}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={790}
              x={475}
              y={373}
            >
              <tspan
                dy="0.71em"
                x={475}
              >
                100
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={790}
              x={80}
              x1={672.5}
              x2={672.5}
              y={365}
              y1={371}
              y2={365}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={790}
              x={672.5}
              y={373}
            >
              <tspan
                dy="0.71em"
                x={672.5}
              >
                150
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={30}
              stroke="#666"
              width={790}
              x={80}
              x1={870}
              x2={870}
              y={365}
              y1={371}
              y2={365}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={30}
              stroke="none"
              textAnchor="middle"
              width={790}
              x={870}
              y={373}
            >
              <tspan
                dy="0.71em"
                x={870}
              >
                200
              </tspan>
            </text>
          </g>
        </g>
      </g>
      <g
        className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
      >
        <line
          className="recharts-cartesian-axis-line"
          fill="none"
          height={360}
          stroke="#666"
          width={60}
          x={20}
          x1={80}
          x2={80}
          y={5}
          y1={5}
          y2={365}
        />
        <g
          className="recharts-cartesian-axis-ticks"
        >
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={360}
              stroke="#666"
              width={60}
              x={20}
              x1={74}
              x2={80}
              y={5}
              y1={50}
              y2={50}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={360}
              stroke="none"
              textAnchor="end"
              width={60}
              x={72}
              y={50}
            >
              <tspan
                dy="0.355em"
                x={72}
              >
                ES6
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={360}
              stroke="#666"
              width={60}
              x={20}
              x1={74}
              x2={80}
              y={5}
              y1={140}
              y2={140}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={360}
              stroke="none"
              textAnchor="end"
              width={60}
              x={72}
              y={140}
            >
              <tspan
                dy="0.355em"
                x={72}
              >
                c#
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={360}
              stroke="#666"
              width={60}
              x={20}
              x1={74}
              x2={80}
              y={5}
              y1={230}
              y2={230}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={360}
              stroke="none"
              textAnchor="end"
              width={60}
              x={72}
              y={230}
            >
              <tspan
                dy="0.355em"
                x={72}
              >
                React
              </tspan>
            </text>
          </g>
          <g
            className="recharts-layer recharts-cartesian-axis-tick"
          >
            <line
              className="recharts-cartesian-axis-tick-line"
              fill="none"
              height={360}
              stroke="#666"
              width={60}
              x={20}
              x1={74}
              x2={80}
              y={5}
              y1={320}
              y2={320}
            />
            <text
              className="recharts-text recharts-cartesian-axis-tick-value"
              fill="#666"
              height={360}
              stroke="none"
              textAnchor="end"
              width={60}
              x={72}
              y={320}
            >
              <tspan
                dy="0.355em"
                x={72}
              >
                CSS
              </tspan>
            </text>
          </g>
        </g>
      </g>
      <g
        className="recharts-layer recharts-bar"
      >
        <g
          className="recharts-layer recharts-bar-rectangles"
          clipPath={null}
        >
          <g
            className="recharts-layer"
          >
            <g
              className="recharts-layer recharts-bar-rectangle"
            />
            <g
              className="recharts-layer recharts-bar-rectangle"
            />
            <g
              className="recharts-layer recharts-bar-rectangle"
            />
            <g
              className="recharts-layer recharts-bar-rectangle"
            />
          </g>
        </g>
      </g>
    </svg>
    <div
      className="recharts-legend-wrapper"
      style={
        Object {
          "bottom": 5,
          "height": "auto",
          "left": 20,
          "position": "absolute",
          "width": 850,
        }
      }
    >
      <ul
        className="recharts-default-legend"
        style={
          Object {
            "margin": 0,
            "padding": 0,
            "textAlign": "center",
          }
        }
      >
        <li
          className="recharts-legend-item legend-item-0"
          style={
            Object {
              "display": "inline-block",
              "marginRight": 10,
            }
          }
        >
          <svg
            className="recharts-surface"
            height={14}
            style={
              Object {
                "display": "inline-block",
                "marginRight": 4,
                "verticalAlign": "middle",
              }
            }
            version="1.1"
            viewBox="0 0 32 32"
            width={14}
          >
            <path
              className="recharts-legend-icon"
              d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
              fill="none"
              strokeWidth={4}
            />
          </svg>
          <span
            className="recharts-legend-item-text"
          >
            Pluralsight Skill IQ
          </span>
        </li>
      </ul>
    </div>
    <div
      className="recharts-tooltip-wrapper"
      style={
        Object {
          "MozTransform": "translate(undefinedpx, undefinedpx)",
          "OTransform": "translate(undefinedpx, undefinedpx)",
          "WebkitTransform": "translate(undefinedpx, undefinedpx)",
          "msTransform": "translate(undefinedpx, undefinedpx)",
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "transform": "translate(undefinedpx, undefinedpx)",
          "visibility": "hidden",
        }
      }
    >
      <div
        className="recharts-default-tooltip"
        style={
          Object {
            "backgroundColor": "#fff",
            "border": "1px solid #ccc",
            "margin": 0,
            "padding": 10,
            "whiteSpace": "nowrap",
          }
        }
      >
        <p
          className="recharts-tooltip-label"
          style={
            Object {
              "margin": 0,
            }
          }
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;
